storage/database.go

```diff
@@ -408,24 +415,39 @@ func (d *db) splayedTick() {
 		return
 	}
 
+	// Cleanup and/or flush if required to cleanup files and/or flush blocks to disk
+	if now := d.nowFn(); d.fsm.ShouldRun(now) {
```

Comment:
We need to move this after ticking because the existing logic is such that the blocks are flushed as soon as possible but no earlier than when they become available. If we run the file ops before ticking, there is a possibility where the blocks just become available but haven't been drained yet so we'll miss those.